DDB Popper v0.2

eclarke12 opened the pull request.

eclarke12 committed:
reroll-initiative redesign for fvtt 0.3.3

eclarke12 committed:
fix typo in function name

Commit:
create enhanced-conditions branch

Commit:
create basic module structure

eclarke12 committed:
remove boilter plate

eclarke12 committed:
basic hook logic

eclarke12 committed:
enhanced conditions basic logic pass

Commit:
chat output wip

eclarke12 committed:
chat message send as token

eclarke12 committed:
remove boiler plate from branch

eclarke12 committed:
alpha release - ready for qa

eclarke12 committed:
additional comments/markup

eclarke12 committed:
restructuring and cleanup

eclarke12 committed:
additional refactoring

eclarke12 committed:
basic functionality complete

eclarke12 committed:
additional comments

eclarke12 committed:
initial rewrite to support settings

eclarke12 committed:
add eslint

eclarke12 committed:
move settings into main class

eclarke12 committed:
working through settings registration bugs

eclarke12 committed:
basic settings logic working

eclarke12 committed:
refactor settings

Commit:
add render hook

eclarke12 committed:
render config hook and settings wip

Commit:
move module init to ready hook. broke settings

Commit:
changed settings to object

eclarke12 committed:
redesign config code based on advice from atropos

Commit:
Merge branch 'reroll-initiative' of https://github.com/death-save/fvtt-modules into reroll-initiative

eclarke12 committed:
create link between config and settings

eclarke12 committed:
Merge branch 'reroll-initiative' of https://github.com/death-save/fvtt-modules into reroll-initiative

eclarke12 committed:
settings refactor. config window wip

Commit:
rescope rri variable into game global

Commit:
wip using game global to hold singletons

Commit:
refactor init

eclarke12 committed:
edit config box

eclarke12 committed:
remove config init from ready hook

eclarke12 committed:
wip link config and setting

Commit:
fixed link between config and setting

eclarke12 committed:
refactoring and cleanup

eclarke12 committed:
basic reroll functionality complete

eclarke12 committed:
refactored hardcode strings to vars where appropriate

eclarke12 committed:
terrible commit message

eclarke12 committed:
refactored settings in rriconfig -- need to re-do

Commit:
wip refactoring config

eclarke12 committed:
wip refactor settings / config

eclarke12 committed:
wip settings rewrite

eclarke12 committed:
wip - need to reassess class instancing

eclarke12 committed:
simplified module version wip

eclarke12 committed:
wip config window

Commit:
wip settings

Commit:
fully functional. cleaned up and commented

Commit:
restructured module folders

Commit:
initial skeleton of function

eclarke12 committed:
move js to folder

eclarke12 committed:
wip trying to figure out best implementation

eclarke12 committed:
Delete reroll-initiative.js

Commit:
add space after arg commas, rem extra setting name

eclarke12 committed:
add incarnateapplicationrender thx @ProNobis#7047

eclarke12 committed:
update to use IncarnateApplicationRender - WIP

eclarke12 committed:
1st working version. wip build extra module function

eclarke12 committed:
fix issue with default settings

eclarke12 committed:
fix issue with settings defaulting on each refresh

eclarke12 committed:
adjust setting logic

eclarke12 committed:
revise settings logic

eclarke12 committed:
change boiler plate

eclarke12 committed:
create config html stub

eclarke12 committed:
redesign to use built in settings config gui

eclarke12 committed:
ignore node modules

eclarke12 committed:
add foundry.js to gitignore

eclarke12 committed:
begin scoping out module

eclarke12 committed:
Merge branch 'reroll-initiative' into combat-utility-belt

eclarke12 committed:
merging in other modules

eclarke12 committed:
Merge remote-tracking branch 'origin/enhanced-conditions' into combat-utility-belt

eclarke12 committed:
move reroll function into main js

eclarke12 committed:
settings logic and basic func for rri and hnn

eclarke12 committed:
add invocations to ready hook

eclarke12 committed:
fix module helper typos

eclarke12 committed:
remove other modules. basic js for button

eclarke12 committed:
basic logic

eclarke12 committed:
template. wip form class

eclarke12 committed:
switch to class design

eclarke12 committed:
fix formapp

eclarke12 committed:
form app wip

eclarke12 committed:
wip handling saving settings

eclarke12 committed:
fix popup logic

eclarke12 committed:
wip settings refactor

eclarke12 committed:
ver0.1 complete

eclarke12 committed:
add readme

eclarke12 committed:
update readme

eclarke12 committed:
fix logic for popup/ddb url detection

eclarke12 committed:
update readme

eclarke12 committed:
update module, create zip

eclarke12 committed:
remove foundry.js

eclarke12 committed:
Merge pull request #1 from eclarke12/ddb-popper

Ddb popper v0.1

eclarke12 committed:
Update Readme.md

eclarke12 committed:
Update Readme.md

eclarke12 committed:
Update Readme.md

eclarke12 committed:
Update Readme.md

eclarke12 committed:
Add files via upload

eclarke12 committed:
Update Readme.md

eclarke12 committed:
fix update in json

eclarke12 committed:
Update module.json

eclarke12 committed:
update ddb-popper.zip

eclarke12 committed:
update readme

eclarke12 committed:
Merge pull request #2 from death-save/ddb-popper

Ddb popper update zip

eclarke12 committed:
Merge branch 'master' into ddb-popper

eclarke12 committed:
Merge pull request #3 from eclarke12/ddb-popper

Ddb popper fixed zip to master

eclarke12 committed:
Update module.json

eclarke12 committed:
fix module.json again

eclarke12 committed:
Add files via upload

eclarke12 committed:
Update Readme.md

eclarke12 committed:
Update Readme.md

eclarke12 committed:
Update Readme.md

eclarke12 committed:
Update Readme.md

eclarke12 committed:
v0.2
add close button logic
remove some hardcode strings

eclarke12 committed:
Merge branch 'ddb-popper' of https://github.com/death-save/fvtt-modules into ddb-popper

eclarke12 committed:
update zip

eclarke12 committed:
Merge branch 'ddb-popper' of https://github.com/death-save/fvtt-modules into ddb-popper

eclarke12 committed:
Merge branch 'ddb-popper' into ddb-popper

eclarke12 committed:
Merge pull request #3 from eclarke12/ddb-popper

Ddb popper v0.2

eclarke12 committed:
update zip

eclarke12 committed:
update readme

eclarke12 committed:
update gitignore

eclarke12 committed:
remove ddb-popper from git tracking

Rawrly85 commented on ddb-popper/ddb-popper.js:
```diff
@@ -0,0 +1,200 @@
+        /**
+         * Remove any existing instances of the button on the sheet
+         */
+        windowHeader.find('.ddb-popup').remove();
```
If it exists already, why do you need to add it again?

eclarke12 replied:
TBH I copied that piece from here: https://github.com/sillvva/foundry-vtt-modules/blob/3fbd15d0792229b22aa9f24359ba40c6af49ecf3/ddb-importer/ddb-importer.js#L27 the main reason would be ensuring that the button isn't duplicated? Or that it has the correct functionality tied in.

Rawrly85 replied:
In your case, you are controlling the adding of th button though? So it shouldn't be duplicated. This is definitely just an optimization change, buy I'm just curious if you saw any bug that caused you to do this.

Rawrly85 commented on ddb-popper/ddb-popper.js:
```diff
@@ -0,0 +1,200 @@
+         * or else simply focus the existing popup
+         */
+        ddbButton.click(async ev => {
+            ev.preventDefault();
```
What's the reason for this? (May just want a comment on why it's required)

eclarke12 replied:
This is a standard method for the `event` interface per MDN: https://developer.mozilla.org/en-US/docs/Web/API/Event/preventDefault it's seemingly standard practice to call this on any non-standard event calls (eg. overwriting context click etc)

Rawrly85 replied:
Yeah, but you usually only do this if it's causing some issue by emitting the event as is. Harmless, but just extra code that I don't see the benefit of.

eclarke12 replied:
For the onClick probably not needed. The context one though is probably necessary to prevent accidental fires of the context menu (I believe it's blocked upstream anyway though).

Rawrly85 commented on ddb-popper/ddb-popper.js:
```diff
@@ -0,0 +1,200 @@
+            ev.preventDefault();
+            const actorDDBURL = await this._getActorDDBURL(data);
+
+            if (actorDDBURL == null || actorDDBURL.length == 0){
```
Is there a case which there is no link to a DDB character sheet and you need to notify the user of this?

eclarke12 replied:
In this case the URL entry form is called, which means the user is informed that the actor is not linked. Is that good enough?

Rawrly85 commented on ddb-popper/ddb-popper.js:
```diff
@@ -0,0 +1,200 @@
+            ev.preventDefault();
+            const actorDDBURL = await this._getActorDDBURL(data);
+
+            new DDBURLEntryForm(actorDDBURL, data, {closeOnSubmit: true}).render(true);
```
Duplicated above, create helper function

eclarke12 replied:
I waffled on this one, because all I'm doing here is instantiating the other module class. Do you really think a helper is the best idea in that scenario?

Rawrly85 commented on ddb-popper/ddb-popper.js:
```diff
@@ -0,0 +1,200 @@
+    constructor(actorDDBURL, data, options){
+        super(data, options);
+        this.data = data;
+        this.actorId = this.data.actor._id;
```
Are these variables defined in the extended class?

You already have access to this.data so I wouldn't create another variable for actorID and rather just access it from this.data. it leave less room for error as you currently are having to make sure both variables are in sync.

eclarke12 replied:
This was something I wrestled with, but since I call actor._id a few times, I thought it might be better to declare it rather than rely on deep dot notation. What do you think?

Rawrly85 replied:
Id create a helper function and call it. The function can do the deep lookup.

eclarke12 committed:
address comments in PR
no longer remove button
don't prevent default on click event
break url form class class into helper
remove actorId initialization

Rawrly85 reviewed (commented):
I don't see an option to change the link once it's set. In the case someone makes a mistake, you'll need to allow for them to fix the link. 

Edit: NVM, I found that right click does this. You should add a tooltip to the button to give context that this can be done (IMO, all of Foundry is missing informative tooltips and it's super annoying from a consumer not knowing what is click able in what way. )

eclarke12 committed:
fix style on img

eclarke12 commented:
> I don't see an option to change the link once it's set. In the case someone makes a mistake, you'll need to allow for them to fix the link.
> 
> Edit: NVM, I found that right click does this. You should add a tooltip to the button to give context that this can be done (IMO, all of Foundry is missing informative tooltips and it's super annoying from a consumer not knowing what is click able in what way. )

I can address this in version 1.1 via an onHover event.

eclarke12 committed:
add title to button a tag

eclarke12 committed:
add tooltip to button
